### PR TITLE
KEYCLOAK-9438: Typescript support for both native and legacy promises

### PR DESCRIPTION
**Pull request by jonaskello**
This PR will add a conditional type so that native promise types can be set as return type from the function of `KeycloakInstance`.

This PR is backwards compatible because by default the legacy promises are used.

To specify that you want native promise types as return types you can now do this:

```typescript
import Keycloak from "keycloak-js";

const keycloak = Keycloak<Keycloak.KeycloakPromiseType>("/keycloak.json");
keycloak
  .init({ onLoad: "login-required", promiseType: "native" })
   .then(authenticated => {
    this.setState({ keycloak: keycloak, authenticated: authenticated });
  });
```

Notice the generic parameter in `Keycloak<Keycloak.KeycloakPromiseType>`, this is what will trigger the native promise types to be used as return types. To use the legacy promise type just use `Keycloak` without any generic params as usual.

### CODE REVIEW

**Review by mhajas — APPROVED** [review]
From my point of view, it looks good. However, I am not very skilled with Typescript, so @ssilvert could you please look into this deeper? I am approving this but I would be happier if Stan approves it before it gets merged.

**Review by ssilvert — APPROVED** [review]
Thanks for the PR.

**Pull request merged**
